AutoROM/AutoROM.py

```diff
@@ -1,175 +1,188 @@
-            installation_dirs.append(mulit_ale_install_dir)
-        else:
-            multi_ale_installed = False
+            return ALEInterface.isSupportedROM(path) is not None
+
+        installation_dirs.append(
+            SupportedPackage(resources.files("ale_py") / "roms", _ale_py_filter)
+        )
+    except ModuleNotFoundError:
+        pass
+
+    # Try and find multi-agent-ale-py
+    try:
+        # Assume all ROMs are supported
+        installation_dirs.append(
+            SupportedPackage(
+                resources.files("multi_agent_ale_py") / "roms", lambda _: True
+            )
+        )
+    except ModuleNotFoundError:
+        pass
+
+    return installation_dirs
+
+
+@click.command()
+@click.option(
+    "--accept-license",
+    is_flag=True,
+    default=False,
+    type=bool,
+    help="Accept license agreement",
+)
+@click.option(
+    "--install-dir",
+    default=None,
+    type=click.Path(exists=True),
+    help="User specified install directory",
+)
+def main(accept_license, install_dir):
+    if install_dir is not None:
+        packages = [SupportedPackage(pathlib.Path(install_dir), lambda _: True)]
     else:
-        mulit_ale_install_dir = None
+        packages = find_supported_packages()
 
-    if not ale_installed and not multi_ale_installed:
-        print("Neither ale_py or multi_ale_py installed, quitting.")
-        quit()
+        if len(packages) == 0:
+            print("Unable to find ale-py or multi-ale-py, quitting.")
+            quit()
 
-    if specific_dir:
-        dir_path = os.path.abspath(os.path.join(specific_dir, "ROM/")) + "/"
-        installation_dirs = [dir_path]
-        ale_install_dir = dir_path
-        mulit_ale_install_dir = dir_path
-
-    __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
-
-    checksum_file = "checksums.txt"
-    ch = open(os.path.join(__location__, checksum_file), "rb")
-    checksum_map = {}
-    for c in ch:
-        payload = c.split()
-        payload[1] = payload[1].decode("utf-8")
-        payload[0] = payload[0].decode("utf-8")
-        checksum_map[payload[0]] = payload[1]
-
-    license_text = ""
-    if ale_installed:
-        license_text += ale_install_dir + "\nfor use with ALE-Py (and Gym)"
-    if ale_installed and multi_ale_installed:
-        license_text += " and also\n\t"
-    if multi_ale_installed:
-        license_text += mulit_ale_install_dir + "\nfor use with Multi-Agent-ALE-py."
-    print("AutoROM will download the Atari 2600 ROMs from",
-        ". \nThey will be installed to\n\t" +
-        license_text + " Existing ROMS will be overwritten.")
-    if not license_accepted:
-        ans = input("\nI own a license to these Atari 2600 ROMs, agree not to "+
-            "distribute these ROMS, and wish to proceed (Y or N). ")
-
-
-        if ans != "Y" and ans != "y":
+    # Get checksums
+    with resources.path("AutoROM", "checksums.txt") as checksum_file:
```

Comment:
Internally, ALE stores the checksums, so ideally, checksums will come from ALE itself, not from AutoROM's file. AutoROM's file would then be removed.

Reply:
Sure, but we don't have some of the ROMs you support. So as of right now, this might be the best solution?

Reply:
I mean, I think it can import multi_agent_ale_py and ale_py and check if one or the other supports the ROM. But multi_agent_ale_py doesn't have the best interface for this right now (it requires loadROM to be called, which is very slow), so I think you are right that right now, it is fine as is.